### PR TITLE
test(avatar): added mocked image tests

### PR DESCRIPTION
**Pull request by anuraghazra**
Closes # 

## 📝 Description

Added skipped tests for Avatar image load/fail cases.

## ⛳️ Current behavior (updates)

> Please describe the current behavior that you are modifying

## 🚀 New behavior

> Please describe the behavior or changes this PR adds

### CODE REVIEW

File: packages/avatar/tests/avatar.test.tsx [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
+      if (loadState === LOAD_IMAGE) {
+        setTimeout(() => {
+          this.onload()
+        }, DELAY)
+      }
+      if (loadState === ERROR_IMAGE) {
+        setTimeout(() => {
+          this.onerror()
+        }, DELAY)
+      }
```
Comment (TimKolberger) [inline packages/avatar/tests/avatar.test.tsx:26]: We could reduce the duplicate lines with a switch statement:

```suggestion
      setTimeout(() => {
        switch (loadState) {
          case LOAD_IMAGE:
            this.onload()
            break
          case ERROR_IMAGE:
            this.onerror()
            break
          default:
            return
        }
      }, DELAY)
```

File: packages/avatar/tests/avatar.test.tsx [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
-import { render, testA11y } from "@chakra-ui/test-utils"
+import { render, testA11y, waitFor } from "@chakra-ui/test-utils"
 import { Avatar, AvatarBadge } from "../src"
 
+const DELAY = 0
+const LOAD_IMAGE = "load"
+const ERROR_IMAGE = "error"
+const orignalImage = window.Image
+
+const mockImage = (loadState: string) => {
```
Comment (TimKolberger) [inline packages/avatar/tests/avatar.test.tsx:10]: There are only two allowed strings for this function, so we can use a more narrow type for `loadState`.

```suggestion
const mockImage = (loadState: "load" | "error") => {
```

**Review by TimKolberger — CHANGES_REQUESTED** [review]
Thank you for this PR @anuraghazra!
I love seeing our codebase getting more bulletproof with tests ✨ 

I requested two small changes in the code. I would be happy to merge this PR when they are fixed.

File: packages/avatar/tests/avatar.test.tsx [hunk 3]
```diff
@@ -7,23 +7,25 @@ const LOAD_IMAGE = "load"
 const ERROR_IMAGE = "error"
 const orignalImage = window.Image
 
-const mockImage = (loadState: string) => {
+const mockImage = (loadState: "load" | "error") => {
   jest.useFakeTimers()
-  ;(window.Image as unknown) = class MockImage {
+  ;(window.Image as unknown) = class {
```
Comment (anuraghazra) [inline packages/avatar/tests/avatar.test.tsx:12]: It was throwing "MockImage" not used eslint error so removed it.

**Comment by anuraghazra** [comment]
Hi @timkindberg I've pushed the requested changes. :)

**Review by TimKolberger — APPROVED** [review]
🚀 thank you for your time @anuraghazra

**Pull request merged**